Merging/Algorithm.cs

```diff
@@ -115,6 +116,10 @@ public static Il2CppReferenceArray<Model> MergeBehaviors(Il2CppReferenceArray<Mo
 
     private static bool IsType<T>(this Type typ)
     {
+        if (typeof(T).IsEnum)
```

Comment:
why was this added exactly, in theory the IsAssignableFrom should have been correct?

Reply:
it threw AccessViolationException so I updated it to use UltimateCrosspathing fixed method

Reply:
ok i guess its a funny il2cpp enum bug yippee